pkg/ingester/ingester.go

```diff
@@ -109,6 +109,13 @@ const (
 	tenantsWithOutOfOrderEnabledStatName   = "ingester_ooo_enabled_tenants"
 	minOutOfOrderTimeWindowSecondsStatName = "ingester_ooo_min_window"
 	maxOutOfOrderTimeWindowSecondsStatName = "ingester_ooo_max_window"
+
+	// Maximum number of TSDB users present on the file system which can be opened in a single process
+	// without concurrency. More precisely, if actual number of TSDB users is lower than this number,
+	// each TSDB is opened in a single process, while WAL replay is done in WALReplayConcurrency concurrent
+	// processes. Otherwise, TSDBs are opened in WALReplayConcurrency concurrent processes, while WAL replay
+	// is done in a single process.
+	maxTSDBOpenWithoutConcurrency = 10
```

Comment:
The description here looks a bit complicated. I would just say that below this number, TSDBs of different tenants are opened sequentially, otherwise concurrently.